drivers/flash/Kconfig.nrf_mram

```diff
@@ -0,0 +1,40 @@
+#
+# Copyright (c) 2024 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+menuconfig SOC_FLASH_NRF_MRAM
+	bool "Nordic Semiconductor flash driver for MRAM"
+	default y
+	depends on DT_HAS_NORDIC_MRAM_ENABLED
+	select FLASH_HAS_DRIVER_ENABLED
+	select FLASH_HAS_PAGE_LAYOUT
+	imply MPU_ALLOW_FLASH_WRITE if ARM_MPU
+	help
+	  Enables Nordic Semiconductor flash driver for MRAM in direct write mode.
+
+	  Note that MRAM words are auto-erased when written to, but writing to a
+	  pre-erased area is faster. Hence, the erase API is not required, but
+	  it can be used to amortize write performance for some use cases.
+
+if SOC_FLASH_NRF_MRAM
+
+config SOC_FLASH_NRF_MRAM_ONE_BYTE_WRITE_ACCESS
+	bool "1-byte write block size"
+	help
+	  By default, write/erase is done on MRAM word-aligned (16-byte) blocks.
+	  This option enables 1-byte block (unaligned) access for write/erase.
+	  This is supported in hardware, but it requires slightly more code to
+	  perform the operations safely.
+
+config SOC_FLASH_NRF_MRAM_LAYOUT_PAGE_SIZE
```

Comment:
This should be moved to the dts or fixed in the driver. This will avoid any confusion:
* there are drivers where this isn't specified in the dts (because the blocks vary in size) in that case it is fixed by the driver.
* there are drivers where this is specified and then it is used for the page layout.

Adding it as a kconfig might result in applications on the same device that get stuck because they where compiled with different kconfig options.

In case where `ONE_BYTE_WRITE_ACCESS` is not selected it should also be a multiple of the `write-block-size`.

Reply:
To be fair, also the `ONE_BYTE_WRITE_ACCESS` should also come from the dts (by setting `write-block-size` equal to 1. But I didn't check how this is done on other drivers.

Reply:
Alright, I'll move both Kconfigs to DT. I realize now that both `write-block-size` and `erase-block-size` seem to be treated as configurable anyway, which is strange, but I can play along. I'd like to come back to this at a future date.